helm-chart/splunk-enterprise/values.yaml

```diff
@@ -0,0 +1,529 @@
+    ## Example: Deploy apps on cluster master instance for local use
+    # appsRepoPollIntervalSeconds: 900
+    # appSources:
+    #   - name: networkApps
+    #     location: networkAppsLoc/
+    #   - name: clusterBase
+    #     location: clusterBaseLoc/
+    #   - name: adminApps
+    #     location: adminAppsLoc/
+    #     scope: local
+    # volumes:
+    #   - name: volume_app_repo_us
+    #     storageType: s3
+    #     provider: aws
+    #     path: bucket-app-framework-us-west-2/idxcAndCmApps/
+    #     endpoint: https://s3-us-west-2.amazonaws.com
+    #     region: us-west-2
+    #     secretRef: s3-secret
+    
+  # Splunk Smartstore configuration
+  # reference: https://github.com/splunk/splunk-operator/blob/develop/docs/SmartStore.md
+  smartstore: {}
+    ## Example: Deploy smartstore to a clustermaster, distributes config to indexer peers
+    # volumes:
+    #   - name: <remote_volume_name>
+    #     path: <remote_volume_path>
+    #     endpoint: https://s3-<region>.amazonaws.com
+    #     secretRef: <secret_store_obj>
+    # indexes:
+    #   - name: <index_name_1>
+    #     remotePath: $_index_name
+    #     volumeName: <remote_volume_name>
+    #   - name: <index_name_2>
+    #     remotePath: $_index_name
+    #     volumeName: <remote_volume_name>
+    #   - name: <index_name_3>
+    #     remotePath: $_index_name
+    #     volumeName: <remote_volume_name>
+  
+  # List volumes to be mounted in all pod containers
+  # reference: https://kubernetes.io/docs/concepts/storage/volumes/
+  volumes: []
+
+  # Full path or URL for a Splunk Enterprise license file
+  # Ignore if deploying a license master instance
+  licenseUrl: ""
+
+  # Full path or URL for one or more default.yml files, separated by commas
+  defaultsUrl: ""
+  
+  # Specify Splunk default.yaml to override intial environment config
+  defaults: {}
+    ## Example: Deploy a multi-site cluster master
+    # splunk:
+    #   site: site1
+    #   multisite_master: localhost
+    #   all_sites: site1,site2,site3
+    #   multisite_replication_factor_origin: 1
+    #   multisite_replication_factor_total: 2
+    #   multisite_search_factor_origin: 1
+    #   multisite_search_factor_total: 2
+    #   idxc:
+    #     search_factor: 2
+    #     replication_factor: 2
+
+  # Full path or URL for one or more defaults.yml files specific to App install, separated by commas
+  defaultsUrlApps: ""
+
+  # List additional environment variables to passed to the Splunk instance container
+  extraEnv: []
+  # - name: 
+  #   value: 
+
+  # Delay in seconds before probing liveness
+  # refernece: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command
+  livenessInitialDelaySeconds: 300
+
+  # Delay in seconds before probing readiness
+  # https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-command
+  readinessInitialDelaySeconds: 10
```

Comment:
These are already defaults. We can remove this? Also, soon these are going to be phased out.
Similarly for other places.

Reply:
in values.yaml file we need to provide all the variables that are in spec. even if they have default values.